oarepo_vocabularies/authorities/providers/openaire_provider.py

```diff
@@ -0,0 +1,254 @@
+        """ Iteratively reach for a key in a nested dictionary """
+        for path in args:
+            if not isinstance(d, dict) or path not in d:
+                return default
+            d = d[path]
+        return d
+    
+    @staticmethod
+    def get_program_from_funding(funding_tree):
+        """ Explicitly search for the first program in the funding tree """
+        if funding_tree == []:
+            return "N/A"
+        if isinstance(funding_tree, list):
+            funder_info = funding_tree[0].items()        
+        else:
+            funder_info = funding_tree.items()
+            
+        for _, value in funder_info:
+            if isinstance(value, dict):
+                if "parent" in value and value["parent"] is not None:
+                    for _, value in value["parent"].items():
+                        if "class" in value:
+                            return value["class"]["$"]
+                if "class" in value:
+                    return value["class"]["$"]
+                
+        return "N/A"
+    
+    @staticmethod
+    def to_vocabulary_item(record):
+        
+        # Parse the record
+        header = record.get("header", {})
+        metadata = record.get("metadata", {})
+        entity = metadata.get("oaf:entity", {})
+        project = entity.get("oaf:project", {})
+        
+        try:
+            relations = project.get("rels", {}).get("rel", [])
+        except KeyError:
+            relations = {}
+        except AttributeError:
+            relations = {}
+        
+        # If there is only one relation, convert it to a list
+        if not isinstance(relations, list):
+            relations = [relations]
+        
+        # Tags (keywords)
+        keywords = project.get("keywords", "")
+        
+        if isinstance(keywords, dict):
+            keywords = keywords.get("$", "")
+        tags = keywords.split(",")
+        
+        # Identifiers
+        identifiers = []
+        
+        identifiers.append({
+            "identifier": header.get("dri:objIdentifier", {}).get("$", ""),
+            "scheme": "dri:objIdentifier"
+        })
+        
+        identifiers.append({
+            "identifier": project.get("originalId", {}).get("$", ""),
+            "scheme": "openaire:originalId"
+        })
+        
+        # Number (code), title (with locale) and acronym
+        number = project.get("code", {}).get("$", "")
+        title = {
+            header.get("locale", {}).get("$", "en")[:2]: project.get("title", {}).get("$", "")
+        }
+        acronym = project.get("acronym", {}).get("$", "")
+        
+        
+        # Funder and according program
+        funding = project.get("fundingtree", [])
+        try:
+            funder = {
+                "id": OpenAIREProvider.dict_get(funding, "funder", "id", "$"),
+                "name": OpenAIREProvider.dict_get(funding, "funder", "name", "$"),
+            }
+        except IndexError:
```

Comment:
Are these errors thrown? dict_get seems to check for path existence before x[...]